### PR TITLE
chore: add header to all requests

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Stacks Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/603461193), the [hosted version](https://pr-1037.app.stacks.engineering), or the [Stacks testnet demo app](https://pr-1037.testnet-demo.stacks.engineering)

Adds a default request header with package version

image

### CODE REVIEW

File: packages/app/src/common/api/fetch.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
+import packageJson from '../../../package.json';
+
 export const withApiServer = (apiServer: string) => (path?: string) =>
   path ? apiServer + path : apiServer;
 
+const defaultHeaders = new Headers([
+  ['x-stx-extension-version', packageJson.version]
```
Comment (aulneau) [inline packages/app/src/common/api/fetch.ts]: How can we ensure these are the same? https://github.com/blockstack/ux/blob/release/canary/packages/app/src/manifest.json#L4

**Review by aulneau — CHANGES_REQUESTED** [review]

File: packages/app/src/common/api/fetch.ts (hunk 1 above)
Comment (aulneau) [reply packages/app/src/common/api/fetch.ts]: https://github.com/blockstack/ux/blob/release/canary/packages/app/webpack.config.js#L238
https://github.com/blockstack/ux/blob/release/canary/packages/app/webpack.config.js#L245
We should use this function to generate the version in the header: https://github.com/blockstack/ux/blob/release/canary/packages/app/webpack.config.js#L41

Comment (kyranjamie) [reply packages/app/src/common/api/fetch.ts]: I could import version from there, instead?

Comment (aulneau) [reply packages/app/src/common/api/fetch.ts]: No I think it has to be a replace kind of thing. In the webpack config, it's generated on build and then with CopyWebpackPlugin, it's replacing the string with a random number appended to the package.json version

Comment (aulneau) [reply packages/app/src/common/api/fetch.ts]: we should use `process.env.VERSION`, it's already added!

Comment (kyranjamie) [reply packages/app/src/common/api/fetch.ts]: it comes out undefined

**Pull request merged**